tests/components/freebox/test_config_flow.py

```diff
@@ -0,0 +1,159 @@
+"""Tests for the Freebox config flow."""
+import asyncio
+from unittest.mock import MagicMock, patch
+
+from aiofreepybox.exceptions import (
+    AuthorizationError,
+    HttpRequestError,
+    InvalidTokenError,
+)
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.freebox.const import DOMAIN
+from homeassistant.config_entries import SOURCE_DISCOVERY, SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.helpers.typing import HomeAssistantType
+
+from tests.common import MockConfigEntry
+
+HOST = "myrouter.freeboxos.fr"
+PORT = 1234
+
+
+@pytest.fixture(name="connect")
+def mock_controller_connect():
+    """Mock a successful connection."""
+    with patch(
+        "homeassistant.components.freebox.config_flow.Freepybox"
+    ) as service_mock:
+        service_mock.return_value.open = MagicMock(return_value=asyncio.Future())
```

Comment:
Import `patch` from `asynctest` and use that to patch the coroutine instead. We can use it for both coroutines and normal functions. Drop `unittest.mock.patch`.

Reply:
I suggest we spec the class to have the patch set `CoroutineMock`s on the instance methods.

Reply:
Oups sorry, didn't get that